openfe/protocols/openmm_utils/omm_settings.py

```diff
@@ -116,7 +116,7 @@ class Config:
     """
     constraint_tolerance = 1e-06
     """Tolerance for the constraint solver. Default 1e-6."""
-    barostat_frequency = 25 * unit.timestep  # todo: IntQuantity
+    barostat_frequency: Optional[FloatQuantity['timestep']] = 25 * unit.timestep  # todo: IntQuantity
```

Comment:
Do we have to add a warning that if this is None, the user shouldn't be running a solvent NPT calculations?

Reply:
There's 4 options here I guess. 
- gas phase & barostat freq set -> warn that this is ignored?
- gas phase & barostat freq None -> ok
- non-gas phase & barostat freq set -> ok
- non-gas phase & barostat freq None -> error out

Reply:
I guess one could make it non-Optional again and then only do option 1 (gas phase & barostat freq set -> warn that this is ignored)?

Reply:
I think that's what we currently have - It's not worth warning if there's nothing that the user can do to avoid the warning.  #712 is asking to make it optional since sometimes it's not relevant.  I guess elsewhere we also have settings that are ignored in certain conditions.

Reply:
@richardjgowers : So you're suggesting leaving it non-Optional?

Reply:
Yeah I think it's worse if it's optional, so keep it as-is and document that this setting is ignored for gas-phase simulations